flytekit/core/artifact.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import annotations
+
+import typing
+from typing import Generic, Optional, TypeVar
+from typing_extensions import Annotated
+
+from flyteidl.artifact import artifacts_pb2 as artifact_idl
+from flyteidl.core.identifier_pb2 import TaskExecutionIdentifier, WorkflowExecutionIdentifier
+from flyteidl.core.literals_pb2 import Literal
+from flyteidl.core.types_pb2 import LiteralType
+from flytekit.core.context_manager import FlyteContextManager
+
+if typing.TYPE_CHECKING:
+    from flytekit.remote.remote import FlyteRemote
+
+T = TypeVar("T")
+
+
+class Artifact(Generic[T]):
+    """
+    Artifact depends on three things, literal type, source, format
+
+    Use one as input to workflow (only workflow for now)
+    df_artifact = Artifact("flyte://a1")
+    remote.execute(wf, inputs={"a": df_artifact})
+    """
+
+    def __class_getitem__(cls, user_type: typing.Type[T], *args, **kwargs) -> typing.Type[T]:
```

Comment:
I do not like this. I like Annotated.

Reply:
this is just swaying away from the style of python and this is what makes things harder